### PR TITLE
Reject set quota for superuser

### PR DESCRIPTION
**Pull request by fanfuxiaoran**

### CODE REVIEW

File: tests/regress/expected/test_role.out [hunk 1]
```diff
@@ -102,6 +102,16 @@ SELECT diskquota.wait_for_worker_new_epoch();
 INSERT INTO b SELECT generate_series(1,100);
 -- expect insert succeed
 INSERT INTO b2 SELECT generate_series(1,100);
+-- superuser is blocked to set quota
+SELECT rolname from pg_roles where rolsuper=true;
+ rolname 
+---------
+ xiwang
```
Comment (beeender) [inline tests/regress/expected/test_role.out:109]: this will fail CI :grin:

Comment (fanfuxiaoran) [reply tests/regress/expected/test_role.out:109]: haha

Comment (beeender) [reply tests/regress/expected/test_role.out:109]: Try this trick?

```
SELECT concat("sys_role_", rolename) from pg_roles where rolsuper=true;
```

And ignore the string begins with `sys_role_`

File: tests/regress/expected/test_role.out [hunk 2]
```diff
@@ -102,6 +102,16 @@ SELECT diskquota.wait_for_worker_new_epoch();
...
+(1 row)
+
+\gset
+select diskquota.set_role_quota(:'rolname', '1mb');
```
Comment (beeender) [inline tests/regress/expected/test_role.out:113]: uh, you have a trick! then I think just `--start-ignore` the above query result could just work?

Comment (fanfuxiaoran) [reply tests/regress/expected/test_role.out:115]: Yeah, this works. Have fixed!

**Review by Sasasu — APPROVED** [review]
add this to `init_file` to make our CI happy

```
-- start_matchsubs
m/^ERROR:  Can not set disk quota from system owner: \c+/
s/^ERROR:  Can not set disk quota from system owner: \c+/ERROR:  Can not set disk quota from system owner:/
-- end_matchsubs
```

**Review by beeender — APPROVED** [review]

**Review by liuxueyang — APPROVED** [review]

**Comment by radarwave** [comment]
As we only limit the system owner, other super users are not affected. Maybe we should change the commit message title to 'system owner' to avoid future confusion.

**Pull request merged**